packages/DropdownMenu/src/DropdownMenu.js

```diff
@@ -49,10 +58,42 @@ const DropdownMenu = props => {
 
   const handleOpenMenu = () => {
     setIsOpen(true);
+    // https://github.com/acl-services/paprika/issues/316
+    // Todo Should focus the first item via an onAfterOpen event callback in popover
+    setTimeout(() => {
+      focusAndSetIndex(0);
+    }, 250);
   };
 
   const extractedChildren = extractChildren(children, ["DropdownMenu.Trigger"]);
 
+  const dropdownLastItemIndex =
+    React.Children.toArray(
+      extractedChildren.children.filter(
+        child =>
+          child.type &&
+          (child.type.displayName === "DropdownMenu.Item" || child.type.displayName === "DropdownMenu.LinkItem")
+      )
+    ).length - 1;
+
+  const onKeyDown = (event, currentIndex) => {
+    if (event.key === "ArrowDown") {
+      const indexToFocus = currentIndex === dropdownLastItemIndex ? 0 : currentIndex + 1;
+      focusAndSetIndex(indexToFocus);
+    } else if (event.key === "ArrowUp") {
+      const indexToFocus = currentIndex === 0 ? dropdownLastItemIndex : currentIndex - 1;
+      focusAndSetIndex(indexToFocus);
+    } else if (event.key === "Home") {
```

Comment:
As discussed with @mikrotron `ArrowRight` and `ArrowLeft` keys should only be relevant if the dropdown Item has nest submenus, otherwise those keys should just close the menu